update changelog for tproxy (#909)
hashicorp/consul-helm: CHANGELOG.md

* CRDs: update the CRD versions from v1beta1 to v1. [[GH-883](https://github.com/hashicorp/consul-helm/pull/883)]
* Enterprise: support applying Consul Enterprise license when security context defaults to non-root users. [[GH-880](https://github.com/hashicorp/consul-helm/pull/880)]
* Sync Catalog: add new `syncCatalog.extraLabels` Helm value for configuring labels on sync catalog pods. [[GH-892](https://github.com/hashicorp/consul-helm/pull/892)]

> * Connect: Support leader election for connect webhook deployment. [[GH-903](https://github.com/hashicorp/consul-helm/pull/903)]

Reviewer: Can we say something that this now allows multiple instances/HA of the connect-inject deployment?

> * Connect: Support leader election for connect webhook deployment. [[GH-903](https://github.com/hashicorp/consul-helm/pull/903)]

Author: ```suggestion
* Connect: Support high availability of the connect-inject deployment. [[GH-903](https://github.com/hashicorp/consul-helm/pull/903)]
```
How about this @thisisnotashwin

Reviewer: <3 love it!

BREAKING CHANGES:
* Minimum Kubernetes versions supported is 1.16+. [[GH-883](https://github.com/hashicorp/consul-helm/pull/883)]
* Connect: `-enable-health-checks-controller`, `-health-checks-reconcile-period`, `-cleanup-controller-reconcile-period` have been removed
  and are no longer supported as the controllers have been replaced by the endpoints controller.

>   The associated helm values have also been removed: `global.connectInject.healthChecks` and `global.connectInject.cleanupController`. [[GH-899](https://github.com/hashicorp/consul-helm/pull/899)]

Reviewer 2: I think this change should be in the context of the helm values rather than command flags since it's for the helm chart. Maybe just rephrase the sentence below saying which helm values have been removed.

* Connect: `-release-name` and `-release-namespace` flags have been added to connect-inject deployment since they are required for endpoints controller to detect consul client pod restart events.

>   [[GH-886](https://github.com/hashicorp/consul-helm/pull/886)]

Reviewer 2: This should not be a breaking change I believe since these flags are not required

Reviewer 2: Oh, wait also we don't need to mention it here since it's not a user-facing change for the helm chart.

* Connect: connect webhook deployment now uses `webhook-cert-manager` to bootstrap the webhook certificates instead of generating them inside of the webhook. [[GH-861](https://github.com/hashicorp/consul-helm/pull/861)]

> 

Reviewer 2: Could we add an explanation of the breaking changes related to requiring a kubernetes service similar to the consul-k8s changelog that includes example service? I think most users look at helm changelog, so it'd be good to move over those details.

This could be a new item saying that this release includes consul-k8s image that requires you to have kubernetes services and then describe what changes users need to make.

> * Connect: connect webhook deployment now uses `webhook-cert-manager` to bootstrap the webhook and controller certificates instead of generating them inside of the webhook. [[GH-861](https://github.com/hashicorp/consul-helm/pull/861)]

Reviewer: I think you can ignore saying controller certificates. Just webhook certificates should be enough (given then controller doesnt use certificates).

> * Connect: connect webhook deployment now uses `webhook-cert-manager` to bootstrap the webhook and controller certificates instead of generating them inside of the webhook. [[GH-861](https://github.com/hashicorp/consul-helm/pull/861)]

Author: ```suggestion
* Connect: connect webhook deployment now uses `webhook-cert-manager` to bootstrap the webhook  certificates instead of generating them inside of the webhook. [[GH-861](https://github.com/hashicorp/consul-helm/pull/861)]
```

Reviewer: perfect!

* CRDs: update the CRD versions from v1beta1 to v1. [[GH-883](https://github.com/hashicorp/consul-helm/pull/883)]
* Enterprise: support applying Consul Enterprise license when security context defaults to non-root users. [[GH-880](https://github.com/hashicorp/consul-helm/pull/880)]
* Sync Catalog: add new `syncCatalog.extraLabels` Helm value for configuring labels on sync catalog pods. [[GH-892](https://github.com/hashicorp/consul-helm/pull/892)]
* Connect: Support high availability of the connect-inject deployment. [[GH-903](https://github.com/hashicorp/consul-helm/pull/903)]

BREAKING CHANGES:
* Minimum Kubernetes versions supported is 1.16+. [[GH-883](https://github.com/hashicorp/consul-helm/pull/883)]

> * Connect: The helm values for healthchecks and cleanup controllers have been removed: `global.connectInject.healthChecks` and `global.connectInject.cleanupController`, as these controllers have been replaced by the endpoints controller. [[GH-899](https://github.com/hashicorp/consul-helm/pull/899)]

Reviewer 2: ```suggestion
* Connect: The Helm values for health checks and cleanup controller have been removed: `connectInject.healthChecks` and `connectInject.cleanupController`, as these controllers have been replaced by the endpoints controller. [[GH-899](https://github.com/hashicorp/consul-helm/pull/899)]
```

* Connect: connect webhook deployment now uses `webhook-cert-manager` to bootstrap the webhook certificates instead of generating them inside of the webhook. [[GH-861](https://github.com/hashicorp/consul-helm/pull/861)]
* Connect: Kubernetes Services are now required for all connect injected applications.
  The Kubernetes service name will be used as the service name to register with Consul unless the annotation `consul.hashicorp.com/connect-service` is provided to the deployment/pod to override this. If using ACLs the ServiceAccountName must match the service name used with Consul.

Note: if you're already using a Kubernetes service, no changes are required.

> Example Service:

Reviewer 2: I think this needs to be indented so that it's still part of that above bullet point. Unless you intended it to be separate?

Author: yeah great catch, thx!

```yaml
---
apiVersion: v1
kind: Service
metadata:
  name: sample-app
spec:
  selector:
    app: sample-app
  ports:
    - port: 80
      targetPort: 9090
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: sample-app
  name: sample-app
spec:
  replicas: 1
  selector:
     matchLabels:
       app: sample-app
   template:
     metadata:
       annotations:
         'consul.hashicorp.com/connect-inject': 'true'
       labels:
         app: sample-app
     spec:
       containers:
       - name: sample-app

>          image: ishustava/fake-service:0.7.0

Reviewer 2: Just so my named image is not in the changelog 😊 
```suggestion
         image: sample-app:0.1.0
```
